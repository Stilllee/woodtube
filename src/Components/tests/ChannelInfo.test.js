import { render, screen, waitFor } from "@testing-library/react";
import { withAllContexts, withRouter } from "../../tests/utils";

import ChannelInfo from "../ChannelInfo";
import React from "react";
import { Route } from "react-router-dom";

describe("ChannelInfo", () => {
  const fakeYoutube = {
    channelImageURL: jest.fn(),
  };
  afterEach(() => fakeYoutube.channelImageURL.mockReset());

  it("올바르게 렌더링", async () => {
    fakeYoutube.channelImageURL.mockImplementation(() => "url");
    const { asFragment } = renderChannelInfo();

    await waitFor(() => screen.getByRole("img"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("URL 없이 렌더링", () => {
    fakeYoutube.channelImageURL.mockImplementation(() => {
      throw new Error("error");
    });

    renderChannelInfo();

    expect(screen.queryByRole("img")).toBeNull();
  });

  it("URL이 있을 때 렌더링", async () => {
    fakeYoutube.channelImageURL.mockImplementation(() => "url");

    renderChannelInfo();

    await waitFor(() => expect(screen.getByRole("img")).toBeInTheDocument());
  });

  function renderChannelInfo() {
    return render(
      withAllContexts(
        withRouter(
          <Route path="/" element={<ChannelInfo id="id" name="channel" />} />
        ),
        fakeYoutube
      )
    );
  }
});
