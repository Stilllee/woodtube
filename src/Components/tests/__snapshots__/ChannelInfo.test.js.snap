// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInfo 올바르게 렌더링 1`] = `
<DocumentFragment>
  <div
    class="flex items-center my-4 mb-8"
  >
    <img
      alt="channel"
      class="w-10 h-10 rounded-full"
      src="url"
    />
    <p
      class="ml-2 text-lg font-medium"
    >
      channel
    </p>
  </div>
</DocumentFragment>
`;
